# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/LIROps.yaml
# Commit: 28feda14fdec
# Full Hash: 28feda14fdec962ea68c26a0570ea69042a5ad27
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2023-08-05 09:19:01
# Regressor Bug: 1692065
# File Overlap Count: 3
# Description:
#   Bug 1692065 - wasm: Add tagging for JSString* in AnyRef. r=yury
#   
#   Before this commit we would need to box a JSString* in WasmValueBox
#   when we received one from JS and converted to an AnyRef.
#   
# ==============================================================================

diff -r 85605041672d -r 28feda14fdec js/src/jit/LIROps.yaml
--- a/js/src/jit/LIROps.yaml	Fri Aug 04 14:06:50 2023 +0000
+++ b/js/src/jit/LIROps.yaml	Fri Aug 04 14:06:50 2023 +0000
@@ -299,7 +299,7 @@
 
 - name: WasmTrapIfNull
   operands:
-    object: WordSized
+    ref: WordSized
   mir_op: true
 
 - name: WasmRefIsSubtypeOfConcrete
@@ -3575,7 +3575,7 @@
 - name: IonToWasmCallI64
   gen_boilerplate: false
 
-- name: WasmBoxValue
+- name: WasmAnyRefFromJSValue
   result_type: WordSized
   operands:
     input: BoxedValue
@@ -3585,6 +3585,11 @@
   operands:
     input: WordSized
 
+- name: WasmAnyRefFromJSString
+  result_type: WordSized
+  operands:
+    input: WordSized
+
 # Constant Simd128
 - name: Simd128
   result_type: WordSized